src/main/appender.cpp

```diff
@@ -375,6 +377,31 @@ void Appender::FlushInternal(ColumnDataCollection &collection) {
 	context->Append(*description, collection);
 }
 
+void Appender::AppendDefault() {
+	if (!defaults[column]) {
+		throw InvalidInputException("Failed to append DEFAULT, this column does not have a DEFAULT value");
+	}
+	auto &default_expr = *defaults[column];
+	if (!default_expr.IsScalar()) {
+		throw InvalidInputException("Only columns with simple DEFAULT values are supported");
```

Comment:
Looking for an example that triggers this, the ConstantBinder has support for a `COLUMN_REF` so I figured it was better to be save than sorry

Reply:
I don't think this can be triggered, so it can be an assertion/internal exception.